controllers/routemonitor/routemonitor.go

```diff
@@ -68,23 +95,19 @@ func (r *RouteMonitorReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 	log.V(2).Info("Response of WasDeleteRequested", "shouldDelete", shouldDelete)
 
 	if shouldDelete {
-		res, err := r.EnsureRouteMonitorAndDependenciesAbsent(ctx, routeMonitor)
+		_, err := r.EnsureMonitorAndDependenciesAbsent(routeMonitor)
 		if err != nil {
 			return utilreconcile.RequeueWith(err)
 		}
-
-		if res.ShouldStop() {
-			return utilreconcile.Stop()
-		}
 		return utilreconcile.Stop()
 	}
 
 	log.V(2).Info("Entering EnsureFinalizerSet")
-	res, err = r.EnsureFinalizerSet(ctx, routeMonitor)
-	if err != nil {
-		return utilreconcile.RequeueWith(err)
-	}
-	if res.ShouldStop() {
+	if r.Common.SetFinalizer(&routeMonitor, consts.FinalizerKey) {
+		_, err := r.Common.UpdateReconciledMonitor(&routeMonitor)
+		if err != nil {
+			return utilreconcile.RequeueWith(err)
+		}
```

Comment:
it's because for most reconcile supplement functions we use a wrapper result as a return value. Meaning the utilreconcile.Result instead of the ctrl.Result. TBH I am also not too big of a fan of it, as I don't understand the gain we got from it. At this point, a translation between the types is happening.
But I believe @georgettica  made some thoughts about it